src/Applications/LDAS_App/util/inputs/ASCAT_sm_mask/ascat_mask_maker.F90

```diff
@@ -0,0 +1,199 @@
+    real, dimension(:), allocatable :: lon, lat, distances
+    real                            :: d_lon, d_lat, ll_lon, ll_lat
+
+    character(400)                  :: fname_in
+
+    ! --------------------------------------------------------------------------------
+    !
+    ! hardwired variables
+
+    ! ASCAT soil moisture mask file from Lindorfer et al 2023 
+    
+    fname_in = '/discover/nobackup/amfox/subsurface_scattering_ASCAT_ERA5_Land.nc'    
+
+    ! Specification of output grid and missing value 
+    
+    d_lon  =    0.1
+    d_lat  =    0.1
+    ll_lon = -180.0
+    ll_lat =  -90.0
+    
+    missing_value = -128
+    
+    ! -------------------------------
+    
+    ! Open the NetCDF file
+    ierr = nf90_open(fname_in, nf90_nowrite, ncid)
+    if (ierr /= nf90_noerr) stop 'Error opening file'
+    
+    ! Data in original mask file are on the 12.5 km fixed Earth grid used for ASCAT (WARP5 grid) and
+    !   stored in the NetCDF file as 1-dimensional arrays of length N_gpi (over land only).
+
+    ! Get the dimension ID
+    ierr = nf90_inq_dimid(ncid, 'gpi', dimid)
+    if (ierr /= nf90_noerr) stop 'Error getting dimension ID'
+    
+    ! Get the length of the dimension
+    ierr = nf90_inquire_dimension(ncid, dimid, len = N_gpi)
+    if (ierr /= nf90_noerr) stop 'Error inquiring dimension'
+    
+    print*, 'N_gpi = ', N_gpi
+
+    ! Allocate the arrays
+    allocate(asc_lon(        N_gpi))
+    allocate(asc_lat(        N_gpi))
+    allocate(cold_mask(      N_gpi))
+    allocate(wet_mask(       N_gpi))
+    allocate(veg_mask(       N_gpi))
+    allocate(subsurface_mask(N_gpi))
+    allocate(combined_mask(  N_gpi))
+
+    ! Get the variable IDs and read the variables
+    ierr = nf90_inq_varid(ncid, 'lon', varid)
+    ierr = nf90_get_var(ncid, varid, asc_lon)
+    ierr = nf90_inq_varid(ncid, 'lat', varid)
+    ierr = nf90_get_var(ncid, varid, asc_lat)
+    ierr = nf90_inq_varid(ncid, 'cold_mask', varid)
+    ierr = nf90_get_var(ncid, varid, cold_mask)
+    ierr = nf90_inq_varid(ncid, 'wet_mask', varid)
+    ierr = nf90_get_var(ncid, varid, wet_mask)
+    ierr = nf90_inq_varid(ncid, 'veg_mask', varid)
+    ierr = nf90_get_var(ncid, varid, veg_mask)
+    ierr = nf90_inq_varid(ncid, 'subsurface_mask', varid)
+    ierr = nf90_get_var(ncid, varid, subsurface_mask)
+
+    ! Close the NetCDF file
+    ierr = nf90_close(ncid)
+    if (ierr /= nf90_noerr) stop 'Error closing file'
+
+    ! Combine the masks (1-dim arrays)
+    where (wet_mask == 1)
+       combined_mask = 1
+    elsewhere
+       combined_mask = subsurface_mask
+    end where
+
+    ! Re-map "combined_mask" from WARP5 input grid to regular lat/lon output grid (2-dim array)
+
+    allocate(lon(int(360.0  / d_lon)))
+    allocate(lat(int(180.0  / d_lat)))
+
+    lon = [(ll_lon + i * d_lon, i = 0, size(lon) - 1)]
+    lat = [(ll_lat + i * d_lat, i = 0, size(lat) - 1)]
```

Comment:
@amfox37 :  What is "lon" and "lat" exactly?  A quick glance at the ASCAT_EUMETSAT reader suggests that "lon" and "lat" are the western/southern boundaries of the grid cell for which the corresponding i,j mask value applies.  Without this cross-referencing, I would have thought that the "lon" and "lat" values here are for a dateline-on-center and pole-on-center grid.  Please double-check to make sure everything is consistent and also documented in the code.

Reply:
Hmm... I'd written this so lat/lon were the western/southern boundaries to correspond with the reader - but I realize now that's probably not what we want as these values are actually being used to calculate the nearest neighbor. And in that case we need the lat/lon for the middle of the grid cell I think?